### PR TITLE
Changed product listing filter links into form buttons to avoid target pages from getting indexed.

### PR DESCRIPTION
**Pull request by Mauricio-Urrego**
### Ticket
- [Prevent links for filters that google can follow. | GACO](https://app.asana.com/0/30156186242982/1190953714418427/f)

### Description
-

### CODE REVIEW

File: src/ProductFilters/WidgetFilterDeliveryTime.php [hunk 1]
```diff
@@ -92,13 +92,14 @@ public function widget($args, $instance) {
       // If the current value is selected the value will not be added.
       if ($values) {
         $link = add_query_arg('delivery_time', implode(',', $values));
```
Comment (sun) [inline src/ProductFilters/WidgetFilterDeliveryTime.php:94]:
```suggestion
```

File: src/ProductFilters/WidgetFilterDeliveryTime.php [hunk 2]
```diff
@@ -92,13 +92,14 @@ public function widget($args, $instance) {
...
+        $value = implode(',', $values);
       }
       else {
         $link = remove_query_arg('delivery_time');
       }
```
Comment (sun) [inline src/ProductFilters/WidgetFilterDeliveryTime.php:99]:
```suggestion
```

File: src/ProductFilters/WidgetFilterDeliveryTime.php [hunk 3]
```diff
@@ -92,13 +92,14 @@ public function widget($args, $instance) {
...
       $chosen = in_array($delivery_time->term_id, $filter_values, TRUE);
       echo sprintf('<li class="%s">', $chosen ? $item_class . ' ' . $item_chosen_class : $item_class);
-      echo sprintf('<a rel="nofollow" href="%s">%s</a></li>', $link, $delivery_time->name);
+      echo sprintf('<button type="submit" class="product-filter-term" formmethod="GET" name="delivery_time" value="%s" formnovalidate>%s</button>', $value, $delivery_time->name);
```
Comment (sun) [inline src/ProductFilters/WidgetFilterDeliveryTime.php:102]: The formnovalidate was just a suggestion, not sure whether it is needed if there is no surrounding form - probably not.
```suggestion
      echo sprintf('<button type="submit" class="product-filter-term" formmethod="GET" name="delivery_time" value="%s">%s</button>', $value, $delivery_time->name);
```

**Review by sun — CHANGES_REQUESTED** [review]
$link appears to be obsolete. I also revised the PR title.

**Comment by Mauricio-Urrego** [comment]
This has some to-do's. It needs to be tied to the checkbox that comes before the button. The button also needs to be able to remove the filter parameter when clicked a second time.

File: src/ProductFilters/WidgetFilterDeliveryTime.php [hunk 4]
```diff
@@ -88,17 +88,11 @@ public function widget($args, $instance) {
...
-      }
-      else {
-        $link = remove_query_arg('delivery_time');
-      }
+      // Submit value as parameter.
+      $value = implode(',', $values);
       $chosen = in_array($delivery_time->term_id, $filter_values, TRUE);
       echo sprintf('<li class="%s">', $chosen ? $item_class . ' ' . $item_chosen_class : $item_class);
-      echo sprintf('<a rel="nofollow" href="%s">%s</a></li>', $link, $delivery_time->name);
+      echo sprintf('<form><a><button type="submit" class="product-filter-term" formmethod="GET" name="delivery_time" value="%s" style="border: none; background-color: #0000;">%s</button></a></form></li>', $value, $delivery_time->name);
```
Comment (fabianmarz) [inline src/ProductFilters/WidgetFilterDeliveryTime.php:95]: 1. Why do you use an a tag in conjunction with a button if the goal is to remove the links? 🤔 
2. This is very hard to read so please put some formatting in :)
3. Don't use inline styles. Use classes instead

Putting a form element around looks like the only way if we don't want to do it with JS. If we do so, why don't you put the action and method on the form element itself? Also what happens, if you want to select multiple filters?

Comment (sun) [reply src/ProductFilters/WidgetFilterDeliveryTime.php:95]: Can we output the wrapping form element only once around all filters?

**Pull request closed without merging**